shelley/chain-and-ledger/cddl-spec/shelley.cddl

```diff
@@ -96,25 +96,53 @@ credential =
   // 2, genesishash
   )
 
-coin = uint
+pool_params = ( #6.258([* keyhash]) ; pool owners
+              , coin                ; cost
+              , unit_interval       ; margin
+              , coin                ; pledge
+              , keyhash             ; operator
+              , $vrf_keyhash        ; vrf keyhash
+              , [credential]        ; reward_account
+              )
 
-epoch = uint
+withdrawal = [credential, amount : uint]
 
-keyhash = $hash
-scripthash = $hash
+full_update = [ protocol_param_update_votes, application_version_update_votes ]
+
+protocol_param_update_votes =
+  { * genesishash => protocol_param_update }
+
+protocol_param_update =
+  { ? 0:  uint               ; minfee A
+  , ? 1:  uint               ; minfee B
+  , ? 2:  uint               ; max block body size
```

Comment:
the above are both `int` in the spec, not usigned

Reply:
hmm, yes.  I don't remember if there was a reason to use `int`. we would never want them to be negative. I think we definitely would not want them to be allowed to be negative on the wire, right?

Reply:
I'll make a note of this and make sure it's as we want on the next (soon) PR.